lib/puppet/functions/binary_file.rb

```diff
@@ -0,0 +1,25 @@
+# Loads a binary file from a module or file system and returns its contents as a Binary.
+# (Documented in 3.x stub)
+#
+# @since 4.8.0
+#
+Puppet::Functions.create_function(:binary_file, Puppet::Functions::InternalFunction) do
+  dispatch :binary_file do
+    scope_param
+    param 'String', :path
+  end
+
+  def binary_file(scope, unresolved_path)
+    path = nil
+    found = Puppet::Parser::Files.find_file(unresolved_path, scope.compiler.environment)
+    if found && Puppet::FileSystem.exist?(found)
```

Comment:
Unless I read the code incorrectly, a file returned from `Puppet::Parser::Files#find_file` is guaranteed to exist (see module.rb:137), so this extra `#exist?` is unnecessary and might be expensive.